frontend/src/employee-mobile-frontend/messages/state.tsx

```diff
@@ -92,9 +93,10 @@ export const MessageContextProvider = React.memo(
 
     const { selectedGroupId } = useSelectedGroup()
 
-    const accounts = useQueryResult(messagingAccountsQuery(unitId ?? ''), {
-      enabled: unitId !== undefined && pinLoginActive
-    })
+    const accounts = useQueryResult(
+      queryOrDefault(messagingAccountsQuery, [])(unitId),
+      { enabled: pinLoginActive }
+    )
```

Comment:
Check how this behaves when `enabled` changes from true to false

Reply:
Cached data is returned. The correct fix would be to include `employeeId` in the query's cache key.

Reply:
Fixed by using `employeeId` in the account query's cache key